src/globus_sdk/client.py

```diff
@@ -78,6 +78,9 @@ class BaseClient:
     #: the scopes for this client may be present as a ``ScopeCollection``
     scopes: ScopeCollection | None = None
 
+    # internal flag to avoid recursion when loading resource_server
+    _is_loading_resource_server: bool = False
+
```

Comment:
I think this is unused? It was something that concerned me a bit, but maybe purely vestigial from an earlier draft.